### PR TITLE
SOC-1027 Don't redirect if we're constructing an article that isn't the current request's article

### PR DESCRIPTION
**Pull request by garthwebb**
https://wikia-inc.atlassian.net/browse/SOC-1027

### CODE REVIEW

File: extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php [hunk 1]
```diff
@@ -916,63 +924,98 @@ static function formatList($comments) {
...
-	 * static entry point for hook
+	 * Static entry point for hook
 	 *
 	 * @param Title $title
 	 * @param Article $article
 	 *
-	 * @static
-	 * @access public
+	 * @return bool
 	 */
```
Comment (garthwebb) [inline extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php]: This is where the actual fix for this ticket takes place.

File: extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php [hunk 2]
```diff
@@ -916,63 +924,98 @@ static function formatList($comments) {
...
-					$query = array();
-
-					if ( $permalink || $commentId !== 0 ) {
-						if( $commentId !== 0 ) {
-							/** bugId:11179 @author: nAndy */
-							$permalink = $commentId;
-						}
+	static public function ArticleFromTitle( Title &$title, &$article ) {
+		// Don't bother checking for redirects if we're not loading this article for the current request
+		if ( !self::isTitleForCurrentRequest( $title ) ) {
```
Comment (garthwebb) [inline extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php:946]: This is the actual fix for this ticket.  Some code earlier the request creates one or more article objects via `Article::newFromTitle` and this code was intercepting it.  If the title in question was a comment it was issuing a redirect to the wall or forum page, with a jump to the comment there.  This makes sense if the user actually wants to load the comment, but if the comment is being loaded to get its info, we shouldn't be redirecting the user.

File: extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php [hunk 3]
```diff
@@ -916,63 +924,98 @@ static function formatList($comments) {
...
+			return false;
+		}
+
+		if ( $title->getPrefixedDBkey() != F::app()->wg->Title->getPrefixedDBkey() ) {
+			return false;
+		}
+	}
+
+	static private function canSetRedirect() {
+		$wg = F::app()->wg;
```
Comment (kvas-damian) [inline extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php]: you could store `Request` in variable while you are not touching `wg` in this method

**Comment by kvas-damian** [comment]
One minor comment. Besides that it looks good to me.

File: extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php [hunk 4]
```diff
@@ -916,63 +924,98 @@ static function formatList($comments) {
...
+		$redirectTitle = Title::newFromText( $parts['title'], MWNamespace::getSubject( $title->getNamespace() ) );
+		if ( empty( $redirectTitle ) ) {
+			return true;
+		}
+
+		$query = [];
+		$commentId = $title->getArticleID();
+		$permalink = F::app()->wg->Request->getInt( 'permalink', 0 );
+
+		// Use comment ID if available - https://wikia.fogbugz.com/f/cases/11179
```
Comment (jsutterfield) [inline extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php:970]: Are we still linking to fogbugz tickets?

Comment (garthwebb) [inline extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php:970]: We still have a reduced access FogBugz account with all our history, so I'm just making a link out of the possibly confusing number given in the [previous comment](https://github.com/Wikia/app/pull/7940/files#diff-e4fe86c4b1039ccf72823b9e6b3135b9L959).

File: extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php (hunk 3 above)
Comment (garthwebb) [inline extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php]: Sounds like a good idea!

**Comment by jsutterfield** [comment]
LGTM

**Pull request merged**